tests/stress/host/host.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include "stress_u.h"
+
+#define ECALL_STRESS_TEST 0
+#define OCALL_STRESS_TEST 1
+#define ECALL_OCALL_STRESS_TEST 2
+// to add: more stress test types
+
+void do_ecall_in_enclave(
+    const char* path,
+    uint32_t flags,
+    int enclave_count,
+    int ecall_count)
+{
+    oe_result_t result;
+    oe_enclave_t* enclave = NULL;
+
+    result = oe_create_stress_enclave(
+        path, OE_ENCLAVE_TYPE_SGX, flags, NULL, 0, &enclave);
+
+    if (result != OE_OK)
+        oe_put_err("oe_create_stress_enclave(): result=%u", result);
+
+    printf("Do ecall in enclave: %d\n", enclave_count);
+    for (int i = 0; i < ecall_count; i++)
+    {
+        if ((result = do_ecall(enclave, i)) != OE_OK)
+        {
+            oe_terminate_enclave(enclave);
+            oe_put_err("test(): result=%u", result);
+        }
+    }
+
+    result = oe_terminate_enclave(enclave);
+    if (result != OE_OK)
+        oe_put_err("oe_terminate_enclave(): result=%u", result);
+}
+
+void do_ecall_by_count_in_same_env(
+    const char* path,
+    uint32_t flags,
+    int enclave_count,
+    int ecall_count)
+{
+    do_ecall_in_enclave(path, flags, enclave_count, ecall_count);
+}
+
+void do_ecall_by_count_in_diff_env_sequential(
+    const char* path,
+    uint32_t flags,
+    int enclave_count,
+    int ecall_count)
+{
+    for (int i = 1; i < enclave_count + 1; i++)
+        do_ecall_in_enclave(path, flags, i, ecall_count);
+}
+
+int main(int argc, const char* argv[])
+{
+    if (argc != 5)
+    {
+        fprintf(
+            stderr,
+            "Usage: %s ENCLAVE_PATH TEST_TYPE ENCLAVE_COUNT ECALL_COUNT\n",
+            argv[0]);
+        exit(1);
+    }
+
+    const uint32_t flags = oe_get_create_flags();
+
+    // load enclave stress test:
+    // realized in create-rapid
+
+    // do ecall / ocall / ecall + ocall stress test:
+    switch (atoi(argv[2]))
+    {
+        // TEST_TYPE=0 - do ecall stress test
+        // 1. do ecall by count in same enclave: 1 enclave, 1000000 ecalls
+        // 2. do ecall by count in diff enclaves sequentially: 100 enclaves, 100000 ecalls for each enclave, sequential
+        case ECALL_STRESS_TEST:
+            if (atoi(argv[3]) == 1)
+                do_ecall_by_count_in_same_env(argv[1], flags, atoi(argv[3]), atoi(argv[4]));
+            if (atoi(argv[3]) > 1)
+                do_ecall_by_count_in_diff_env_sequential(argv[1], flags, atoi(argv[3]), atoi(argv[4]));
+            // to add more do ecall stress tests:
+            // 3. do ecall by count in diff enclaves parallelly
```

Comment:
Note for this future test: you're going to have to do a dynamic EPC detection to figure out how many enclaves you can support in parallel because you cannot assume that the test machine this code lands on has sufficient EPC for a fix enclave count like (e.g. 100 as in test 2)

Reply:
Yes, will add dynamic EPC detection in future test cases here.